contracts/utils/EternalStorage.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title EternalStorage
+ * @dev This contract holds all the necessary state variables to carry out the storage of any contract.
+ */
+contract EternalStorage {
```

Comment:
@Foivos @milapsheth I don't think that we need to use Eternal Storage. It is an outdated pattern and it does not have 100% storage collision proof

We should calculate the storage slots or use name-spaced storage instead:
https://eips.ethereum.org/EIPS/eip-7201#reference-implementation